cclient/serialization/serializer.h

```diff
@@ -0,0 +1,135 @@
+
+  size_t (*get_size_deserialize_get_inclusion_state_response)(
+      const serializer* const, const char* const toDesirialize);
+
+  // get_neighbors_response
+  void (*serialize_get_neighbors_response)(const serializer* const,
+                                           const get_neighbors_res_t* const obj,
+                                           char* out);
+
+  void (*deserialize_get_neighbors_response)(const serializer* const,
+                                             const char* const obj,
+                                             get_neighbors_res_t* out);
+
+  size_t (*get_size_serialize_get_neighbors_response)(
+      const serializer* const, const get_neighbors_res_t* const toSerialize);
+
+  size_t (*get_size_deserialize_get_neighbors_response)(
+      const serializer* const, const char* const toDesirialize);
+
+  // get_node_info_response
+  void (*serialize_get_node_info_response)(const serializer* const,
+                                           const get_node_info_res_t* const obj,
+                                           char* out);
+
+  void (*deserialize_get_node_info_response)(const serializer* const,
+                                             const char* const obj,
+                                             get_node_info_res_t* out);
+
+  size_t (*get_size_serialize_get_node_info_response)(
+      const serializer* const, const get_node_info_res_t* const toSerialize);
+
+  size_t (*get_size_deserialize_get_node_info_response)(
+      const serializer* const, const char* const toDesirialize);
+
+  // get_tips_response
+  void (*serialize_get_tips_response)(const serializer* const,
+                                      const get_tips_res_t* const obj,
+                                      char* out);
+
+  void (*deserialize_get_tips_response)(const serializer* const,
+                                        const char* const obj,
+                                        get_tips_res_t* out);
+
+  size_t (*get_size_serialize_get_tips_response)(
+      const serializer* const, const get_tips_res_t* const toSerialize);
+
+  size_t (*get_size_deserialize_get_tips_response)(
+      const serializer* const, const char* const toDesirialize);
+
+  // get_transactions_to_approve_response
+  void (*serialize_get_transactions_to_approve_response)(
+      const serializer* const,
+      const get_transactions_to_approve_res_t* const obj, char* out);
+
+  void (*deserialize_get_transactions_to_approve)(
+      const serializer* const, const char* const obj,
+      get_transactions_to_approve_res_t* out);
+
+  size_t (*get_size_serialize_get_transactions_to_approve)(
+      const serializer* const,
+      const get_transactions_to_approve_res_t* const toSerialize);
+
+  size_t (*get_size_deserialize_get_transactions_to_approve)(
+      const serializer* const, const char* const toDesirialize);
+
+} serializer_vtable;
+
+typedef struct serializer_base {
+  serializer_vtable* vtable;
```

Comment:
let's not make this a pointer? leads to slightly more memory being used, but should be a bit more efficient